### PR TITLE
Update windows doc to mention rust-lld linker error

### PR DESCRIPTION
**Pull request by IceSentry**
Release Notes:

- N/A

## Description

When using rust-lld it's possible to get a `STATUS_ACCESS_VIOLATION` error at compile time. I added a bit of information about it in the build guide for windows to recommend using a different linker when building `zed`.

### CODE REVIEW

**Review by osiewicz — APPROVED** [review]
LGTM, thanks!

File: docs/src/development/windows.md [hunk 1]
```diff
@@ -86,3 +86,11 @@ Before reporting the issue, make sure that you have the latest rustc version wit
 ### Cargo errors claiming that a dependency is using unstable features
 
 Try `cargo clean` and `cargo build`.
+
+### `STATUS_ACCESS_VIOLATION`
+
+This error can happen if you are using the "rust-lld.exe" linker. Consider trying a different linker.
+
+If you are using a global config. Consider moving zed to a nested directory and add a `.cargo/config.toml` with a custom linker config in the parent directory.
```
Comment (maxdeviant) [inline docs/src/development/windows.md:94]:
```suggestion
If you are using a global config, consider moving the Zed repository to a nested directory and add a `.cargo/config.toml` with a custom linker config in the parent directory.
```

File: docs/src/development/windows.md [hunk 2]
```diff
@@ -86,3 +86,11 @@ Before reporting the issue, make sure that you have the latest rustc version wit
...
+
+See this issue for more information https://github.com/zed-industries/zed/issues/12041
```
Comment (maxdeviant) [inline docs/src/development/windows.md:96]: Can we make this a proper Markdown link?

**Review by maxdeviant — CHANGES_REQUESTED** [review]

File: docs/src/development/windows.md (hunk 2 above)
Comment (IceSentry) [reply docs/src/development/windows.md:96]: Do you prefer `<url>` or `[url](url)`?

Comment (maxdeviant) [reply docs/src/development/windows.md:96]: > Do you prefer `<url>` or `[url](url)`?

`[url](url)`

Comment (IceSentry) [reply docs/src/development/windows.md]: done

**Review by maxdeviant — APPROVED** [review]

**Pull request merged**